core/tests/test_box.h

```diff
@@ -0,0 +1,388 @@
+
+    box_free(a);
+}
+
+// Create a new box and check if the min and max are set at the values we wanted
+void test_box_new_2(void) {
+    Box* a = box_new_2(0.0f, 0.0f, 0.0f, 1.0f, 1.0f, 1.0f);
+
+    float3 limitCheck = {0.0f, 0.0f, 0.0f};
+    TEST_CHECK(float3_isEqual(&a->min, &limitCheck, EPSILON_0_0001_F) == true);
+    TEST_CHECK(float3_isEqual(&a->max, &limitCheck, EPSILON_0_0001_F) == false);
+
+    float3_set(&limitCheck, 1.0f, 1.0f, 1.0f);
+    TEST_CHECK(float3_isEqual(&a->min, &limitCheck, EPSILON_0_0001_F) == false);
+    TEST_CHECK(float3_isEqual(&a->max, &limitCheck, EPSILON_0_0001_F) == true);
+
+    box_free(a);
+}
+
+// Create a box with set min and max then copy them into a new box. Then we check the values in the new box
+void test_box_new_copy(void) {
+    Box* a = box_new_2(3.0f, 5.0f, 2.0f, 13.0f, 15.0f, 12.0f);
+    Box* b = box_new_copy(a);
+
+    float3 limitCheck = {3.0f, 5.0f, 2.0f};
+    TEST_CHECK(float3_isEqual(&a->min, &limitCheck, EPSILON_0_0001_F) == true);
+    TEST_CHECK(float3_isEqual(&b->min, &limitCheck, EPSILON_0_0001_F) == true);
+    TEST_CHECK(float3_isEqual(&a->max, &limitCheck, EPSILON_0_0001_F) == false);
+    TEST_CHECK(float3_isEqual(&b->max, &limitCheck, EPSILON_0_0001_F) == false);
+
+    float3_set(&limitCheck, 13.0f, 15.0f, 12.0f);
+    TEST_CHECK(float3_isEqual(&a->min, &limitCheck, EPSILON_0_0001_F) == false);
+    TEST_CHECK(float3_isEqual(&b->min, &limitCheck, EPSILON_0_0001_F) == false);
+    TEST_CHECK(float3_isEqual(&a->max, &limitCheck, EPSILON_0_0001_F) == true);
+    TEST_CHECK(float3_isEqual(&b->max, &limitCheck, EPSILON_0_0001_F) == true);
+
+    TEST_CHECK(float3_isEqual(&a->min, &b->min, EPSILON_0_0001_F));
+    TEST_CHECK(float3_isEqual(&a->max, &b->max, EPSILON_0_0001_F));
+
+    box_free(a);
+    box_free(b);
+}
+
+// Create a box with set min and max and set a new center for the box in a float3.
+// We now set the new center of the box and check if the values are corrects.
+void test_box_set_bottom_center_position(void) {
+    Box* a = box_new_2(3.0f, 5.0f, 2.0f, 13.0f, 15.0f, 12.0f);
+    float3 setcenter = {10.0f, 10.0f, 10.0f};
+
+    box_set_bottom_center_position(a, &setcenter);
+
+    float3 limitCheck = {5.0f, 10.0f, 5.0f};
+    TEST_CHECK(float3_isEqual(&a->min, &limitCheck, EPSILON_0_0001_F) == true);
+    TEST_CHECK(float3_isEqual(&a->max, &limitCheck, EPSILON_0_0001_F) == false);
+
+    float3_set(&limitCheck, 15.0f, 20.0f, 15.0f);
+    TEST_CHECK(float3_isEqual(&a->min, &limitCheck, EPSILON_0_0001_F) == false);
+    TEST_CHECK(float3_isEqual(&a->max, &limitCheck, EPSILON_0_0001_F) == true);
+
+    box_free(a);
+}
+
+// Create 2 boxes with differents values and check both of their center 
+void test_box_get_center(void) {
+    Box* a = box_new_2(0.0f, 0.0f, 0.0f, 10.0f, 10.0f, 10.0f);
+    Box* b = box_new_2(8.0f, 15.0f, 1.0f, 16.0f, 35.0f, 2.0f);
+    float3* getcenter = float3_new(0.0f, 0.0f, 0.0f);
```

Comment:
Don't allocate `float3`s on the heap when you can do it on the stack. Let's keep things simple.

See how I corrected the previous functions:
```c
float3 limitCheck = {0.0f, 0.0f, 0.0f};
```

You can do this in the rest of the functions below.